src/Features/Core/Portable/SolutionCrawler/IdleProcessor.cs

```diff
@@ -39,6 +48,11 @@ internal abstract class IdleProcessor
         protected abstract Task WaitAsync(CancellationToken cancellationToken);
         protected abstract Task ExecuteAsync();
 
+        /// <summary>
+        /// Will be called in a serialized fashion (i.e. never concurrently).
```

Comment:
note: this is hte stated invariant of hte system from before.  i've done no analysis to ensure it's actually true.